processing/src/main/java/org/apache/druid/math/expr/ExprEval.java

```diff
@@ -844,8 +849,12 @@ public final ExprEval castTo(ExpressionType castTo)
             case STRING:
               return ExprEval.ofStringArray(value == null ? null : new Object[] {value.toString()});
           }
+        case COMPLEX:
```

Comment:
need a `break;` above this so `ARRAY` doesn't fall through?

Reply:
good catch, fixed and added tests